pkg/tfshim/sdk-v2/provider_options.go

```diff
@@ -81,6 +81,9 @@ func getProviderOptions(opts []providerOption) (providerOptions, error) {
 
 // Selectively opt-in resources that pass the filter to using PlanResourceChange. Resources are
 // identified by their TF type name such as aws_ssm_document.
+//
+// Deprecated: this no longer has any effect.
+// TODO[pulumi/pulumi-terraform-bridge#2593] clean up deprecation.
 func WithPlanResourceChange(filter func(tfResourceType string) bool) providerOption { //nolint:revive
 	return func(opts providerOptions) (providerOptions, error) {
 		opts.planResourceChangeFilter = filter
```

Comment:
To make it clearer that this does nothing, we can remove the implementation here.
```suggestion
```